File: src/platform/sim/conf.py
# Configuration file for the linux backend

specific_files = "boot.s utils.s hostif_%s.c platform.c host.c" % comp[ 'cpu' ].lower()
ldscript = "i386.ld"
  
# override default optimize settings (-Os is broken right now)
comp.Replace(OPTFLAGS = ['-O0'])
comp.AppendUnique(CCFLAGS = '-g'])

# Prepend with path
specific_files = " ".join( [ "src/platform/%s/%s" % ( platform, f ) for f in specific_files.split() ] )
ldscript = "src/platform/%s/%s" % ( platform, ldscript )

# Standard GCC Flags
comp.Append(CCFLAGS = ['-ffunction-sections','-fdata-sections','-fno-strict-aliasing','-Wall'])
comp.Append(LINKFLAGS = ['-nostartfiles','-nostdlib','-T',ldscript,'-Wl,--gc-sections','-Wl,--allow-multiple-definition'])
#comp.Append(ASFLAGS = ['-x','assembler-with-cpp','-c','-Wall','$_CPPDEFFLAGS'])
comp.Append(LIBS = ['c','gcc','m'])

TARGET_FLAGS = ['-march=i386','-mfpmath=387','-m32']

comp.Prepend(CCFLAGS = [TARGET_FLAGS,'-fno-builtin','-fno-stack-protector'])
comp.Prepend(LINKFLAGS = [TARGET_FLAGS,'-Wl,-e,start','-Wl,-static'])
comp['AS'] = toolset[ 'asm' ]  # Need to force toolset
comp.Prepend(ASFLAGS = ['-felf'])

# Toolset data
tools[ 'sim' ] = {}

# Programming function for i386 (not needed, empty function)
def progfunc_dummy( target, source, env ):
  print "Run the simulator and enjoy :)"
  
tools[ 'sim' ][ 'progfunc' ] = progfunc_dummy

File: src/platform/sim/conf.lua
-- Configuration file for the linux (sim) backend

specific_files = sf( "boot.s utils.s hostif_%s.c platform.c host.c", comp.cpu:lower() )
local ldscript = "i386.ld"
  
-- Override default optimize settings
delcf{ "-Os", "-fomit-frame-pointer" }
addcf{ "-O0", "-g" }

-- Prepend with path
specific_files = utils.prepend_path( specific_files, sf( "src/platform/%s", platform ) )
local ldscript = sf( "src/platform/%s/%s", platform, ldscript ) 

-- Standard GCC flags
addcf{ '-ffunction-sections', '-fdata-sections', '-fno-strict-aliasing', '-Wall' }
addlf{ '-nostartfiles', '-nostdlib', '-T', ldscript, '-Wl,--gc-sections', '-Wl,--allow-multiple-definition' }
addlib{ 'c','gcc','m' }
addm({"_READ_WRITE_RETURN_TYPE=_ssize_t"})

local target_flags = { '-march=i386','-mfpmath=387','-m32' }

addcf{ target_flags, '-fno-builtin', '-fno-stack-protector' }
addlf{ target_flags, '-Wl,-e,start', '-Wl,-static' }
addaf{ '-felf' }

-- Need to force the assembler command as we're using nasm
ascmd = "nasm -felf -o $(TARGET) $(FIRST)"
-- Also tell the builder that we don't need dependency checks for assembler files
builder:set_asm_dep_cmd( false )

-- Toolset data
tools.sim = {}

-- Programming function for i386 (not needed, empty function)
tools.sim.progfunc = function( target, deps )
  print "Run the simulator (./run_elua_sim.sh) and enjoy :) Linux only."
  return 0
end

-- Add the programming function explicitly for this target
tools.sim.pre_build = function()
  local t = builder:target( "#phony:prog", { exetarget }, tools.sim.progfunc )
  builder:add_target( t, "build eLua firmware image", { "prog" } )
end

